scripts/workflow-parameter-extraction.py

```diff
@@ -0,0 +1,152 @@
+            trace_struct_declarations(rhs, struct_type_info)
+    elif isinstance(obj, WDL.Decl):
+        trace_struct_declarations(obj.expr, struct_type_info)
+    elif isinstance(obj, WDL.Expr.Base):
+        for child in obj.children:
+            trace_struct_declarations(child, struct_type_info)
+
+    if isinstance(obj, WDL.Expr.Ident) and isinstance(obj.type, WDL.Type.Array):
+        var_name = str(doc.workflow.name) + "." + str(obj.name)
+        for param in obj.type.parameters:
+            if isinstance(param, WDL.Type.StructInstance):
+                struct_fields = []
+                for struct_member in param.members.values():
+                    struct_fields.append({'inner': str(struct_member).lower()})
+                struct_type_info[var_name] = {'inner': 'array', 'fields': {'inner': 'object', 'fields': struct_fields}}
+    elif (isinstance(obj, WDL.Expr.Ident) and isinstance(obj.type, WDL.Type.StructInstance)
+          and not isinstance(obj.referee, WDL.Tree.Scatter)):
+        var_name = str(doc.workflow.name) + "." + str(obj.name)
+        struct_fields = []
+        for struct_member in obj.type.members.values():
+            struct_fields.append({'inner': str(struct_member).lower()})
+        struct_type_info[var_name] = {'inner': 'object', 'fields': struct_fields}
+    return struct_type_info
+
+
+def pull_inputs():
+    workflow_items = {}
+    for item in doc.workflow.available_inputs or []:
+        var_name = str(doc.workflow.name) + "." + str(item.name)
+        var_type = str(item.value.type)
+        if var_type.endswith('?'):
+            workflow_items[var_name] = {'inner': var_type[:-1].lower(), 'is': 'optional'}
+        else:
+            workflow_items[var_name] = var_type.lower()
+            
+    # Now recursively traverse entire document hunting for Structs or Arrays of Struct
+    struct_items = trace_struct_declarations(doc)
+    workflow_items.update(struct_items)
+    
+    return workflow_items
+    
+    
+def pull_outputs():
+    workflow_items = {}
+    for item in doc.workflow.outputs or []:
+        var_name = str(doc.workflow.name) + "." + str(item.name)
+        var_type = str(item.type)
+        workflow_items[var_name] = var_type.lower()
+    return workflow_items
+
+
+# For testing purposes
+def write_out():
+    if output_config_path:
+        output_path = output_config_path
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+    else:
+        name_index = input_wdl_path.rfind('/') + 1
+        output_path = input_wdl_path[:name_index] + "_config"
+    
+    with open(output_path, "w") as output_file:
+        workflow_inputs = pull_inputs()
+        workflow_outputs = pull_outputs()
+        
+        workflow = {'language': 'WDL_1_0',
+                    'outputs': workflow_outputs,
+                    'parameters': workflow_inputs,
+                    'workflow': repr(doc.source_text)}
+        
+        for key, value in workflow.items():
```

Comment:
Are you serialising this JSON manually? You can use `import json` and `json.dump(workflow, output_file)`.